File: js/factorial.js
const fs = require('fs');

const start = process.hrtime.bigint();
function factorial(n) {
    let result = BigInt(1);
    for (let i = 2; i <= n; i++) {
        result *= BigInt(i);
    }
    return result;
}
const n = 10000;
factorial(n);
const end = process.hrtime.bigint();

// Convertir a milisegundos
const executionTime = Number(end - start) / 1e6;

console.log(`Tiempo de ejecución: ${executionTime.toFixed(2)}`);
fs.appendFileSync('resultadoJS.txt', `Tiempo de ejecución: ${executionTime.toFixed(2)} ms\n`, 'utf8');

File: ruby/factorial.rb
def factorial(n)
  result = 1
  (2..n).each do |i|
    result *= i
  end
  result
end

n = 10000
start_time = Time.now # Obtener el tiempo de inicio

# Calcular el factorial
factorial(n)

end_time = Time.now # Obtener el tiempo de finalización
execution_time = (end_time - start_time) * 1000 # Convertir a milisegundos

# Guardar en un archivo de texto
file_path = "resultadoRuby.txt"
File.open(file_path, 'w') do |file|
  file.write(format('%.2f', execution_time))
end

# Imprimir también en consola
puts format('%.2f', execution_time)
